Assets/MixedRealityToolkit/Interfaces/InputSystem/IMixedRealityRaycastProvider.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Physics;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// Implements the Raycast Provider for handling raycasts into the scene.
+    /// </summary>
+    public interface IMixedRealityRaycastProvider : IMixedRealityDataProvider
+    {
+        /// <summary>
+        /// Simple raycasts each physics <see cref="Microsoft.MixedReality.Toolkit.Physics.RayStep"/>.
+        /// </summary>
+        /// <returns>Whether or not the raycast hit something.</returns>
+        bool RaycastSimplePhysicsStep(RayStep step, LayerMask[] prioritizedLayerMasks, out MixedRealityRaycastHit hitInfo);
```

Comment:
To keep the interface simple, I would remove the versions that don't receive maxDistance. If we think it is a common use case to want to use the step length as max distance, we can implement that as an extension. See this for a reference: https://stackoverflow.com/a/2567204/3876392

Reply:
About the methods names, I would rather shorten them to `Raycast()` and `SphereCast()`.

Reply:
The overloaded variants aren't used currently, so I'll remove.